### ISSUE TITLE
Issue with "<datalist>: The HTML Data List element": Promoting a poor usage

### ISSUE THREAD

**Issue by alattalatta** [mdn/content#5334]
MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist

#### What information was incorrect, unhelpful, or incomplete?

The "[Customizing datalist styles](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist#customizing_datalist_styles)" example is promoting a poor usage of `<datalist>`. It's not keyboard navigable, and screen readers (at least VoiceOver) doesn't announce it properly. Tested with Safari. Plain `<div>`s fit better here.

But before all that, it's not even working. With Chrome and Firefox, `onblur`'s hiding mechanism makes it not clickable. With Safari, `<option>`s are not visible at all.

#### Specific section or headline?

[Customizing datalist styles](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist#customizing_datalist_styles)

#### What did you expect to see?

No such usage?

#### Did you test this? If so, how?

Described above.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/html/element/datalist`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/element/datalist/index.html
* Last commit: https://github.com/mdn/content/commit/2e0349b80f88b4b924459485c6d7eafd325097a8
* Document last modified: 2021-05-02T05:49:34.000Z

</details>

**Comment by alattalatta** [comment]
Duplicate of #4944.
